### PR TITLE
feat(client): add support for buffer notifications

### PR DESCRIPTION
**Pull request by chemzqm**
Make node-client works with PR https://github.com/neovim/neovim/pull/7917

### CODE REVIEW

File: src/host/index.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export class Host {
 
   // Route incoming request to a plugin
   async handlePlugin(method: string, args: any[]) {
+    if (method.startsWith('nvim_buf_')) return
```
Comment (justinmk) [inline src/host/index.ts]: Just for reference, our plan is to reserve the `nvim_` prefix for events. Also, all events will have the `_event` suffix.

So maybe it makes sense to ignore `nvim_`, not just `nvim_buf_` ?

Comment (chemzqm) [reply src/host/index.ts]: Sure.

File: src/api/Buffer.ts [hunk 2]
```diff
@@ -112,6 +112,16 @@ export class Buffer extends BaseApi {
     });
   }
 
+  /* Turn on live updates */
+  attach(send_buffer: boolean) {
```
Comment (billyvg) [inline src/api/Buffer.ts]: Can you change `send_buffer` to be `sendBuffer`?

**Comment by billyvg** [comment]
I just read up on https://github.com/neovim/neovim/pull/7917#issuecomment-392744195 - I'm going to change the interface for this and remove `attach`/`detach` here (`attach` also doesn't work, missing an arg).

**Pull request merged**